config/provider.go

```diff
@@ -153,6 +155,13 @@ func GetProvider() *tjconfig.Provider {
 	}
 
 	pc.ConfigureResources()
+
+	for _, r := range pc.Resources {
```

Comment:
Why do we do this "defaulting" after custom configurations? If we move this to before custom `.Configure` functions are run, in those custom configuration functions, we would be able to override any unsuitable defaulting performed.

Reply:
Added a comment to code:

This function runs after the special configurations were applied. However, if some references were configured in the configuration files, the reference generator does not override them.